skyportal/handlers/api/photometric_series.py

```diff
@@ -0,0 +1,1764 @@
+import os
+import base64
+import traceback
+
+import pandas as pd
+import arrow
+
+from astropy.time import Time
+
+import sqlalchemy as sa
+from sqlalchemy import func
+from sqlalchemy.sql.expression import case
+import conesearch_alchemy as ca
+
+from baselayer.app.access import permissions  # , auth_or_token
+from marshmallow.exceptions import ValidationError
+
+from baselayer.app.env import load_env
+from baselayer.log import make_log
+
+from ..base import BaseHandler
+from ...enum_types import ALLOWED_BANDPASSES
+from ...models.photometric_series import (
+    PhotometricSeries,
+    verify_data,
+    infer_metadata,
+    verify_metadata,
+)
+from ...models.group import Group
+from ...models.stream import Stream
+from ...models.obj import Obj
+from ...models.instrument import Instrument
+from ...models.followup_request import FollowupRequest
+from ...models.assignment import ClassicalAssignment
+
+
+_, cfg = load_env()
+
+body_schema_docstring = """
+  content:
+    application/json:
+      schema:
+        type: object
+        properties:
+          data:
+            type: string or dict
+            description: |
+              The data to upload. Can be a string or a dict.
+              If a dict (i.e., a json object) will assume each
+              key is a column name and each value is a list of
+              values for that column.
+              That dictionary will be passed into a pandas
+              DataFrame constructor, so the keys must be valid
+              and the length of each value must be the same.
+              If a string, will be converted to a bytes array
+              and de-serialized by the pandas HDF5 reader.
+              Use the HDFStore to create a file that includes
+              a single key/group with the photometric data.
+              Additional information, including any of the
+              parameters specified below, can be stored in
+              the HDFStore as well, inside the attributes
+              under the key "metadata".
+              In any case the DataFrame must have the following
```

Comment:
Maybe should add explicitly the list of required and allowed optional columns here?

Reply:
We don't require anything except "mjd" and "mag" or "flux", and there could be additional columns that have meanings like "fluxerr" or "magerr" that are used mainly for plotting. 
The users are free to add any column they wish and that is stored just for record-keeping, skyportal doesn't do anything with those.